src/Microsoft.OData.Core/UriParser/Binders/MetadataBinder.cs

```diff
@@ -354,7 +355,15 @@ protected virtual QueryNode BindStringLiteral(StringLiteralToken stringLiteralTo
         /// <returns>The bound In token.</returns>
         protected virtual QueryNode BindIn(InToken inToken)
         {
-            InBinder inBinder = new InBinder(this.Bind);
+            Func<QueryToken, QueryNode> InBinderMethod = (queryToken) =>
+             {
+                 if (queryToken.Kind == QueryTokenKind.Literal)
+                 {
+                     return LiteralBinder.BindInLiteral((LiteralToken)queryToken);
+                 }
+                 return this.Bind(queryToken);
+             };

```

Comment:
shall we move it into a separate protected virtual method?

Reply:
The entire InBind method is virtual and protected. There will always be an option to override the entire method instead of creating another virtual method.